docs/learning/backend/Workspace.md

```diff
@@ -1,475 +1,222 @@
+```ts
+[[include:WorkspaceExamples.saveSettingDictionary]]
+```
 
-Containers can each have independent access rights, and users and applications are granted permissions to read, write, create, etc. by authenticating their identity and then obtaining a container-specific (usually expiring) *shared access signature* token (a `sasToken`) from the storage authority.
+Note that modifying the iModel settings requires obtaining an exclusive write lock on the entire iModel. Ordinary users should never perform this kind of operation - only administrators.
 
-Cloud-based `WorkspaceContainer`s provide a mechanism for storing and retrieving `WorkspaceDb`s through a secure, reliable, and highly available cloud api.
+The next time we open the iModel, the new settings dictionary will automatically be loaded, and we can query its settings:
 
-Data stored in cloud-based `WorkspaceDb`s:
+```ts
+[[include:WorkspaceExamples.QuerySettingDictionary]]
+```
 
-- can be versioned
-- can have fine-grained access permissions, or may be publicly accessible
-- can be accessed directly from cloud storage without pre-downloading
-- is automatically *cached* locally for fast access
-- can be fully downloaded for offline use
-- is automatically synched when changes are made
+The "hardinessRange" setting is obtained from the iModel's settings dictionary, while the "defaultTool" falls back to the value defined in `IModelHost.appWorkspace.settings`.
 
-The `WorkspaceContainer` apis abstract the cloud storage implementation, so the may be configured to use any cloud storage system (e.g. Azure, AWS, Google, etc.)
+## Workspace resources
 
-A [WorkspaceContainer]($backend) is a special type of cloud container that (only) holds [WorkspaceDb]($backend)s. [WorkspaceDb]($backend)s are databases that hold [workspace resources](#workspace-resources).
+"Resources" are bits of data that an application depends on at run-time to perform its functions. The kinds of resources can vary widely from one application to another, but some common examples include:
+- [TextStyle]($common)s and fonts used when placing [TextAnnotation]($common)s.
+- [GeographicCRS]($common)es used to specify an iModel's spatial coordinate system.
+- Images that can be used as pattern maps for [Texture]($backend)s.
+- [SettingsDictionary]($backend)s defining reusable settings.
 
-Conceptually, you can picture the hierarchy like this:
+It might be technically possible to store resources in [Setting]($backend)s, but doing so would present significant disadvantages:
+- Some resources, like images and fonts, may be defined in a binary format that is inefficient to represent using JSON.
+- Some resources, like geographic coordinate system definitions, must be extracted to files on the local file system before they can be used.
```

Comment:
CRS's (and fonts) are not extracted to files. They are read directly from blobs.

Reply:
I thought the whole thing with files was that CSMap needs to do fopen. If that's not the case, do we have any real-world examples of file resources?

Reply:
No, I added virtual methods to CSMap to read data from SQLite. There are no uses of files in core. I expect apps will need them though for backwards compatibility, unfortunately. Maybe we don't publish them until someone asks?

Reply:
Adjusted.